app/addons/documents/index-results/actions.js

```diff
@@ -96,7 +104,7 @@ function (app, FauxtonAPI, ActionTypes, Stores, Documents, SidebarActions) {
             collection: collection,
             query: options.queryCode,
             textEmptyIndex: 'No Results Found!',
-            bulkCollection: Documents.BulkDeleteDocCollection
+            bulkCollection: bulkCollection
```

Comment:
i really got into trouble here.

the collection is stores in the store, and received by `indexResultsStore.getCollection()` -- the same collection is put into the paginationStore to handle page updates.

both stores and actions (paginationStore, index-resultStore, paginationAction, paginationStore) alter the collection and it really got hairy here. 

in this file i tried the approach of the full roundtrip with the bulkDeleteCollection in `deleteSelected ` (scroll down) - but not happy with it either (easier to reason about, but too much code too change right now as it is used this way everywhere) - apart from that i am happy with it, maybe just do the work 1 time, what do you think?

any other alternatives?